cores/esp32/esp32-hal-periman.h

```diff
@@ -72,21 +109,33 @@ typedef bool (*peripheral_bus_deinit_cb_t)(void * bus);
 
 const char* perimanGetTypeName(peripheral_bus_type_t type);
 
-// Sets the bus type and bus handle for given pin.
-bool perimanSetPinBus(uint8_t pin, peripheral_bus_type_t type, void * bus);
+// Sets the bus type, bus handle, bus number and bus channel for given pin.
+bool perimanSetPinBus(uint8_t pin, peripheral_bus_type_t type, void * bus, int8_t bus_num, int8_t bus_channel);
 
 // Returns handle of the bus for the given pin if type of bus matches. NULL otherwise
 void * perimanGetPinBus(uint8_t pin, peripheral_bus_type_t type);
 
 // Returns the type of the bus for the given pin if attached. ESP32_BUS_TYPE_MAX otherwise
 peripheral_bus_type_t perimanGetPinBusType(uint8_t pin);
 
+// Returns the bus number or unit of the bus for the given pin if set. -1 otherwise
+int8_t perimanGetPinBusNum(uint8_t pin);
+
+// Returns the bus channel of the bus for the given pin if set. -1 otherwise
+int8_t perimanGetPinBusChannel(uint8_t pin);
+
 // Sets the peripheral destructor callback. Used to destroy bus when pin is assigned another function
 bool perimanSetBusDeinit(peripheral_bus_type_t type, peripheral_bus_deinit_cb_t cb);
 
 // Check if given pin is a valid GPIO number 
 bool perimanPinIsValid(uint8_t pin);
 
+// Sets extratype for non Init bus
```

Comment:
`extra type` and maybe try to explain it slightly better? :D

Reply:
Better now? :D

Reply:
better :P